tf_quant_finance/volatility/bachelier_tf.py

```diff
@@ -0,0 +1,248 @@
+# -*- coding: utf-8 -*-
+"""
+""
+Created on Fri Nov 22 15:22:13 2019
+
+'Copyright 2020 Joerg Kienitz
+
+'Redistribution and use in source and binary forms, with or without modification,
+'are permitted provided that the following conditions are met:
+
+'1. Redistributions of source code must retain the above copyright notice,
+'this list of conditions and the following disclaimer.
+
+'2. Redistributions in binary form must reproduce the above copyright notice,
+'this list of conditions and the following disclaimer in the documentation
+'and/or other materials provided with the distribution.
+
+'3. Neither the name of the copyright holder nor the names of its contributors
+'may be used to endorse or promote products derived from this software without
+'specific prior written permission.
+
+'THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+'"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+'THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+'ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+'FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+'(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+'LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+'ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+'OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+'THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@author: Joerg Kienitz
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+# straight fwd implementation of the Bachelier pricing
+# there is a version with just one call to exp !!
+
+    
+def option_price(spots, 
+                 strikes, 
+                 volatilities, 
+                 expiries,
+                 rates,
+                 is_call_options=None,
+                 dtype = None,
+                 name = None):
+    """ Compute the Bachelier price for a batch of European options.
+    
+  ## References:
+  [1] Kienitz, J. "Interest Rate Derivatives Explained I", Plagrave McMillan (2014) p.119
+  [2] https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3428994
```

Comment:
Could you please wright a full reference and add a link. See, e.g., [here](https://github.com/google/tf-quant-finance/blob/1bcc6af9a7a3f7e51ec3aaee04b54add63dbbc12/tf_quant_finance/math/interpolation/cubic/cubic_interpolation.py#L70)